### PR TITLE
(1228a) Risks and needs Section 7

### PR DESCRIPTION
**Pull request by jsrobertson**
## Context

https://trello.com/c/FUjaZCgu/1228-add-oasys-section-7-page-lifestyle-and-associates-m

## Changes in this PR
- New `Lifestyle` type (name may change depending on what it gets called in the API)
- New client for calling `/oasys/:prisonNumber/lifestyle`
- New `OasysService` method to call client and return data, or not.
- New `RisksAndNeedsController` method for lifestyle and associates page.
- Add Section 7 - Lifestyle and associates to Risk and needs page navigation

## Screenshots of UI changes

localhost_3000_refer_referrals_fc70b23a-47bf-4015-9c1e-18b5b8bffcf7_risks-and-needs_lifestyle-and-associates

## Release checklist

[Release process documentation](../doc/how-to/perform-a-release.md)

As part of our continuous deployment strategy we must ensure that this work is
ready to be released once merged.

### Pre-merge

- [ ] There are changes required to the Accredited Programmes API for this change to work...
  - [ ] ... and they have been released to production already

### Post-merge

- [ ] This adds, extends or meaningfully modifies a feature...
  - [ ] ... and I have written or updated an end-to-end test for the happy path in the [Accredited Programmes E2E repo](https://github.com/ministryofjustice/hmpps-accredited-programmes-e2e)
- [ ] This makes new expectations of the API...
  - [ ] ... and I have notified the API developer(s) of changes to the contract tests (Pact), or the API is already compliant
- [ ] [Manually approve](../doc/how-to/perform-a-release.md#releasing-to-the-preprod-environment) release to preprod
- [ ] [Manually approve](../doc/how-to/perform-a-release.md#releasing-to-the-production-environment) release to prod

### CODE REVIEW

File: server/controllers/shared/risksAndNeedsController.test.ts [hunk 1]
```diff
@@ -94,6 +98,55 @@ describe('RisksAndNeedsController', () => {
...
 
+  describe('lifestyleAndAssociates', () => {
+    it('renders the lifestyle and associates page with the correct response locals', async () => {
+      const lifestyle = lifestyleFactory.build({ activitiesEncourageOffending: '0 - No problems' })
+      const reoffendingSummaryListRows = [{ key: { text: 'key-one' }, value: { text: 'value one' } }]
+
+      mockLifestyleAndAssociatesUtils.reoffendingSummaryListRows.mockReturnValue(reoffendingSummaryListRows)
+
+      when(oasysService.getLifestyle).calledWith(username, person.prisonNumber).mockResolvedValue(lifestyle)
+      when(oasysService.getOffenceDetails).calledWith(username, person.prisonNumber).mockResolvedValue(null)
```
Comment (yndajas) [inline server/controllers/shared/risksAndNeedsController.test.ts:109]: Don't think we need this second `when` mock (same below)?

File: server/views/referrals/show/risksAndNeeds/lifestyleAndAssociates.njk [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        attributes: {
+          "data-testid": "reoffending-summary-list"
+        }
+      },
+      rows: reoffendingSummaryListRows
+    }) }}
+  {% else %}
+    {% set noOffenceDetailsString = "No lifestyle data found in OASys. Add lifestyle data to OASys to see them here." %}
+
+    {{ keylessSummaryCard("Section 7.2 - Reoffending", bodyText=noOffenceDetailsString, testId="no-lifestyle-data-summary-card") }}
```
Comment (yndajas) [inline server/views/referrals/show/risksAndNeeds/lifestyleAndAssociates.njk:25]: `noLifestyleDataString`, and elsewhere we've used the main section for the title of this summary card, so "Section 7 - Lifestyle and associates" here?

File: server/routes/assess.ts [hunk 3]
```diff
@@ -20,6 +20,7 @@ export default function routes(controllers: Controllers, router: Router): Router
   get(assessPaths.show.programmeHistory.pattern, referralsController.programmeHistory())
   get(assessPaths.show.sentenceInformation.pattern, referralsController.sentenceInformation())
 
+  get(assessPaths.show.risksAndNeeds.lifestyleAndAssociates.pattern, risksAndNeedsController.lifestyleAndAssociates())
```
[the remark this answers is not in the document]
Comment (jsrobertson) [reply server/routes/assess.ts:23]: Good spot.  I've split the commit and reordered.

File: server/views/referrals/show/risksAndNeeds/lifestyleAndAssociates.njk [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+        attributes: {
+          "data-testid": "reoffending-summary-list"
+        }
+      },
+      rows: reoffendingSummaryListRows
+    }) }}
+  {% else %}
+    {% set noLifestyleDataString = "No lifestyle data found in OASys. Add lifestyle data to OASys to see them here." %}
+
+    {{ keylessSummaryCard("Section 7 - Lifestyle and associates", bodyText=noOffenceDetailsString, testId="no-lifestyle-data-summary-card") }}
```
Comment (jsrobertson) [inline server/views/referrals/show/risksAndNeeds/lifestyleAndAssociates.njk:25]: Updating to `noLifestyleDataString` 🙄

**Pull request merged**